File: benchs/ruby/sum_array.rb
class Array
  def sum
`    total = 0;
    for (var i=0;i < this.length;i++) {
      total += this[i];
    }

    return total;
`
  end
end

`window.rbTests.sum_array = function() {`
  a = []
  `for (var i=0;i < 1000;i++) {`
    `a.push(1);`
  `}`
  a.sum
`}`

File: benchs/js/sum_array.js
Array.prototype.sum = function() {
  total = 0;
  for (var i=0;i < this.length;i++) {
    total += this[i];
  }

  return total;
};

window.jsTests.sum_array = function() {
  var a = [];
  for (var i=0;i < 1000;i++) {
    a.push(1);
  }

  a.sum();
};